File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/reference/part-two-python/src/overview.py
from grafana_foundation_sdk.builders import common, logs, stat, text, timeseries
from grafana_foundation_sdk.models.common import (
    GraphDrawStyle,
    LegendDisplayMode,
    LegendPlacement,
    TooltipDisplayMode,
    StackingMode,
)
from grafana_foundation_sdk.models.dashboard import DynamicConfigValue

from .catalog import Service
from .common import (
    stat_panel,
    instant_prometheus_query,
    loki_datasource_ref,
    prometheus_datasource_ref,
    text_panel,
    loki_query,
    timeseries_panel,
)


def version_stat(service: Service) -> stat.Panel:
    return (
        stat_panel()
        .title("Version")
        .with_target(instant_prometheus_query('app_infos{service="%s"}' % service.name))
        .transparent(True)
        .datasource(prometheus_datasource_ref())
        .reduce_options(
            common.ReduceDataOptions()
            .values(False)
            .calcs(["last"])
            .fields("/^version$/")
        )
    )


def description_text(service: Service) -> text.Panel:
    return text_panel(service.description).transparent(True)


def logs_volume_timeseries(service: Service) -> timeseries.Panel:
    return (
        timeseries_panel()
        .title("Logs volume")
        .with_target(
            loki_query(
                'sum by (level) (count_over_time({service="%s", level=~"$logs_level"} |~ "$logs_filter" [$__auto]))'
                % service.name
            ).legend_format("{{ level }}")
        )
        .datasource(loki_datasource_ref())
        .stacking(common.StackingConfig().mode(StackingMode.NORMAL))
        .transparent(True)
        .tooltip(common.VizTooltipOptions().mode(TooltipDisplayMode.MULTI))
        .legend(
            common.VizLegendOptions()
            .display_mode(LegendDisplayMode.LIST)
            .placement(LegendPlacement.RIGHT)
            .show_legend(True)
        )
        .draw_style(GraphDrawStyle.BARS)
        .override_by_name(
            "INFO",
            [
                DynamicConfigValue(
                    id_val="color", value={"mode": "fixed", "fixedColor": "green"}
                ),
            ],
        )
        .override_by_name(
            "ERROR",
            [
                DynamicConfigValue(
                    id_val="color", value={"mode": "fixed", "fixedColor": "red"}
                ),
            ],
        )
    )

File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/reference/part-two-golang/overview.go
package main

import (
	"fmt"

	"github.com/grafana/grafana-foundation-sdk/go/common"
	"github.com/grafana/grafana-foundation-sdk/go/dashboard"
	"github.com/grafana/grafana-foundation-sdk/go/stat"
	"github.com/grafana/grafana-foundation-sdk/go/text"
	"github.com/grafana/grafana-foundation-sdk/go/timeseries"
)

func versionStat(service Service) *stat.PanelBuilder {
	return statPanel().
		Title("Version").
		WithTarget(
			instantPrometheusQuery(fmt.Sprintf("app_infos{service=\"%s\"}", service.Name)),
		).
		Transparent(true).
		Datasource(prometheusDatasourceRef()).
		ReduceOptions(common.NewReduceDataOptionsBuilder().
			Values(false).
			Calcs([]string{"last"}).
			Fields("/^version$/"),
		)
}

func descriptionText(service Service) *text.PanelBuilder {
	return textPanel(service.Description).
		Transparent(true)
}

func logsVolumeTimeseries(service Service) *timeseries.PanelBuilder {
	return timeseriesPanel().
		Title("Logs volume").
		WithTarget(
			lokiQuery(fmt.Sprintf("sum by (level) (count_over_time({service=\"%s\", level=~\"$logs_level\"} |~ \"$logs_filter\" [$__auto]))", service.Name)).
				LegendFormat("{{level}}"),
		).
		Stacking(common.NewStackingConfigBuilder().Mode(common.StackingModeNormal)).
		Transparent(true).
		Datasource(lokiDatasourceRef()).
		Tooltip(common.NewVizTooltipOptionsBuilder().Mode(common.TooltipDisplayModeMulti)).
		Legend(common.NewVizLegendOptionsBuilder().
			Placement(common.LegendPlacementRight).
			ShowLegend(true).
			DisplayMode(common.LegendDisplayModeList),
		).
		DrawStyle(common.GraphDrawStyleBars).
		OverrideByName("INFO", []dashboard.DynamicConfigValue{
			{
				Id:    "color",
				Value: map[string]any{"mode": "fixed", "fixedColor": "green"},
			},
		}).
		OverrideByName("ERROR", []dashboard.DynamicConfigValue{
			{
				Id:    "color",
				Value: map[string]any{"mode": "fixed", "fixedColor": "red"},
			},
		})
}
